extmod/vfs_posix_file.c

```diff
@@ -92,18 +91,17 @@ mp_obj_t mp_vfs_posix_file_open(const mp_obj_type_t *type, mp_obj_t file_in, mp_
         }
     }
 
-    o->base.type = type;
-
     mp_obj_t fid = file_in;
 
+    int fd;
     if (mp_obj_is_small_int(fid)) {
-        o->fd = MP_OBJ_SMALL_INT_VALUE(fid);
-        return MP_OBJ_FROM_PTR(o);
+        fd = MP_OBJ_SMALL_INT_VALUE(fid);
+    } else {
+        const char *fname = mp_obj_str_get_str(fid);
+        MP_HAL_RETRY_SYSCALL(fd, open(fname, mode_x | mode_rw, 0644), mp_raise_OSError(err));
     }
-
-    const char *fname = mp_obj_str_get_str(fid);
-    int fd;
-    MP_HAL_RETRY_SYSCALL(fd, open(fname, mode_x | mode_rw, 0644), mp_raise_OSError(err));
+    mp_obj_vfs_posix_file_t *o = m_new_obj_with_finaliser(mp_obj_vfs_posix_file_t);
```

Comment:
If this raises an exception, e.g. due to out of memory, the open file handle will leak.

Reply:
the type needs to be set down where it was originally as it gets changed by the switch in the function, but yes pre-seeing fd to -1 was an earlier version of the fix I had. I was thinking it's safer to not leak the object at all if the open fails but hadn't considered the opposite issue!

Reply:
Thanks I've updated as such